src/main/java/org/wildfly/client/config/ConfigurationXMLStreamReader.java

```diff
@@ -725,13 +726,57 @@ default Expression getExpressionAttributeValue(int index, Expression.Flag... fla
         final String attributeValue = getAttributeValue(index);
         if (attributeValue == null) {
             return null;
+        } else if (attributeValue.startsWith("env.")) {
+            String envVar = System.getenv().get(attributeValue.substring(4));
+
+            if (envVar.startsWith("ENC:")) {
+                return resolveEncryptedExpression(envVar, flags);
+            } else {
+                return Expression.compile(envVar, flags);
+            }
+        } else if (attributeValue.contains("env") || attributeValue.contains("ENV")) {
+            String envVar = replaceNonAlphanumericByUnderscoresAndMakeUpperCase(attributeValue);
+
+            if (envVar.startsWith("ENC:")) {
+                return resolveEncryptedExpression(envVar, flags);
+            } else {
+                return Expression.compile(envVar, flags);
+            }
+        } else if (attributeValue.startsWith("prop.")) {
+            String propertyValue = System.getProperty(attributeValue.substring(5));
+
+            if (propertyValue.startsWith("ENC:")) {
+                return resolveEncryptedExpression(propertyValue, flags);
+            } else {
+                return Expression.compile(propertyValue, flags);
+            }
+        } else if (attributeValue.contains("prop") || attributeValue.contains("PROP")) {
+            String propertyValue = System.getProperty(replaceNonAlphanumericByUnderscoresAndMakeUpperCase(attributeValue));
+
+            if (propertyValue.startsWith("ENC:")) {
+                return resolveEncryptedExpression(propertyValue, flags);
+            } else {
+                return Expression.compile(propertyValue, flags);
+            }
+        } else if (attributeValue.contains("ENC:")) {
+            return resolveEncryptedExpression(attributeValue, flags);
         } else try {
             return Expression.compile(attributeValue, flags);
         } catch (IllegalArgumentException ex) {
             throw msg.expressionParseException(ex, getAttributeName(index), getLocation());
         }
     }
 
+    default Expression resolveEncryptedExpression(String attributeValue,  Expression.Flag... flags) throws ConfigXMLParseException {
+        Iterator resolverProviderIterator = ServiceLoader.load(ResolverProvider.class).iterator();
```

Comment:
Raw types should never be used, due to their propensity towards introducing avoidable bugs; this should be `Iterator<ResolverProvider>` instead.

Reply:
Fixed!